### ISSUE TITLE
Show total energy from LG air conditioner

### ISSUE THREAD

**Issue by scoop643** [ollo69/ha-smartthinq-sensors#530]
My LG air conditioner shows in the app the actual power and the total energy. In HA i can only see the actual power. Is it possible to also show the total energy so i can show it on the energy dashboard?

If I need to test something; that is possible.

**Comment by wesgeorge** [comment]
This is supported across a number of different devices, and there are currently separate feature requests for several of them. Might make sense to consolidate this into one feature request to generically support energy usage in any entity that offers it, since it's likely the same in the API.

**Comment by SiboutVanLoo** [comment]
You can, create a template sensor to convert power reading to kwh.

**Comment by scoop643** [comment]
> You can, create a template sensor to convert power reading to kwh.

That is correct, but they are not accurate at all. I've tried several.  But they all deviate from the measurement in the air conditioner.

**Comment by SiboutVanLoo** [comment]
In what way?
Up our down?
A lot, a little,...
The app says, actual data may differ,...
The only way to acurately define power usage is measuring.

Op za 1 jul. 2023 06:35 schreef scoop643 ***@***.***>:

**Comment by SiboutVanLoo** [comment]
The Thinq app says “Real power usage can vary”.
My guess is the app makes an estimate, there was an issue with standby power use, but that’s been rectified to 5w.
The template looks at the 5w standby power use and counts it in, and the app doesn’t. (The app only accounts for use power, as a reference. Don’t look at it as true data. but more as a, "good to know tool" to compare weeks/months)

My guess is the template sensor is more accurate than the app, i monitor power use on the main meter p1 port, and realtime power use of the AC in HA correlates with what i see there.
The Thinq app gives me lower readings. (sure, there’s no point that LG would scale the power data to real numbers, less is more in marketing termes, right?) :-D 

Do you have a way of measuring the real time power use? (And check with HA sensor data.)

**Comment by skinkie** [comment]
Just implemented this with a helper (Riemann sum integral sensor). Works great as "Individual Device" in the energy dashboard now.
